[HUDI-509] Renaming code in sync with cWiki restructuring

vinothchandar opened the pull request:
## *Tips*
- *Thank you very much for contributing to Apache Hudi.*
- *Please review https://hudi.apache.org/contributing.html before opening a pull request.*

## What is the purpose of the pull request

This PR renames classes,configs in Hudi to align with the new terminologies defined in the cWiki design and architecture page.

## Brief change log

 - Storage Type replaced with Table Type (remaining instances)
 - View types replaced with query types;
 - ReadOptimized view referred as Snapshot Query
 - TableFileSystemView sub interfaces renamed to BaseFileOnly and Slice Views
 - HoodieDataFile renamed to HoodieBaseFile
 - Hive Sync tool will register RO tables for MOR with a `_ro` suffix
 - Datasource/Deltastreamer options renamed accordingly
 - Support fallback to old config values as well, so migration is painless
 - Config for controlling _ro suffix addition
 - Integration test fixed accordingly

## Verify this pull request

This change is covered by existing tests; integration tests have been redone where needed

## Committer checklist

 - [ ] Has a corresponding JIRA in PR title & commit
 
 - [ ] Commit message is descriptive of the change
 
 - [ ] CI is green

 - [ ] Necessary doc changes done or have another open PR
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

lamber-ken commented on hudi-common/src/main/java/org/apache/hudi/common/model/HoodieBaseFile.java:
```diff
@@ -27,21 +27,21 @@
- * Hoodie data file.
+ * Hoodie base file.
  */
-public class HoodieDataFile implements Serializable {
```
This means ORC support is coming 👍

vinothchandar commented:
@n3nash are you able to review this?  or any other committer really.. (wish there was a gh team we could mention.. )

n3nash commented on hudi-common/src/main/java/org/apache/hudi/common/table/SyncableFileSystemView.java:
```diff
@@ -18,12 +18,15 @@
+
 /**
- * A consolidated file-system view interface exposing both realtime and read-optimized views along with
+ * A consolidated file-system view interface exposing both full and basefile only views along with
```
s/full/complete?

vinothchandar replied:
okay .. `complete` sounds better

n3nash commented on hudi-hive/src/main/java/org/apache/hudi/hive/HiveSyncTool.java:
```diff
@@ -53,30 +53,44 @@
-  private final HoodieHiveClient hoodieHiveClient;
-  public static final String SUFFIX_REALTIME_TABLE = "_rt";
+  public static final String SUFFIX_SNAPSHOT_TABLE = "_rt";
+  public static final String SUFFIX_READ_OPTIMIZED_TABLE = "_ro";
```
Not sure if it's the best user-experience to augment the table name with a postfix for all tables..

vinothchandar commented:
@n3nash Addressed comments.. Responded to the renaming issue. Let me know what you think and we can hopefully merge after CI passes? 

I will follow up with site doc changes.

bhasudha commented on hudi-common/src/main/java/org/apache/hudi/common/table/timeline/dto/DataFileDTO.java:
```diff
@@ -36,22 +36,22 @@
   private long fileLen;
 
-  public static HoodieDataFile toHoodieDataFile(DataFileDTO dto) {
+  public static HoodieBaseFile toHoodieDataFile(DataFileDTO dto) {
```
@vinothchandar Do we want to rename the method as well to 'toHoodieBaseFile' ?

n3nash replied:
+1

bhasudha commented on hudi-common/src/main/java/org/apache/hudi/common/table/timeline/dto/DataFileDTO.java:
```diff
@@ -36,22 +36,22 @@
   }
 
-  public static DataFileDTO fromHoodieDataFile(HoodieDataFile dataFile) {
+  public static DataFileDTO fromHoodieDataFile(HoodieBaseFile dataFile) {
```
'fromHoodieBaseFile' instead ?

n3nash replied:
+1

bhasudha commented on hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java:
```diff
@@ -249,10 +249,10 @@ private void ensurePartitionLoadedCorrectly(String partition) {
    * @param statuses List of File-Status
    */
-  private Stream<HoodieDataFile> convertFileStatusesToDataFiles(FileStatus[] statuses) {
+  private Stream<HoodieBaseFile> convertFileStatusesToDataFiles(FileStatus[] statuses) {
```
convertFileStatusesToBaseFiles instead ? No strong opinion on these. I think it should be okay to leave it as is since the return type is indicative already.

n3nash replied:
+1

bhasudha commented:
LGTM overall. Left minor comments. Was just thinking about them. Feel free to ignore if its too much refactoring.

vinothchandar commented:
@bhasudha good catch. I cleaned up what you pointed out.. but there are still tons of variables named `dataFile` instead of `baseFile`. I created a JIRA for later.. 

@n3nash On the suffix, I believe this is the least path of change for existing users. I believe most users will be using teh _rt table with MOR.. So we can't rename that or drop the suffix there... Its better to add it to the RO table to make it clear.. I even added a flag to hive sync so that existing users can skip _ro suffix if needed..

Ideally we had no suffix on the snapshot table and only did a special _ro suffix... or even drop the _ro talbe altogether.. but this is a much larger change.. We can pursue this on top of what this PR does anyway
